internal/yunikorn/sync_nodes_int_test.go

```diff
@@ -0,0 +1,121 @@
+package yunikorn
+
+import (
+	"context"
+	"time"
+
+	"github.com/G-Research/unicorn-history-server/internal/database/repository"
+	"github.com/G-Research/unicorn-history-server/internal/model"
+	"github.com/G-Research/yunikorn-core/pkg/webservice/dao"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type SyncNodesTestSuite struct {
+	suite.Suite
+	pool *pgxpool.Pool
+	repo *repository.PostgresRepository
+}
+
+func (ss *SyncNodesTestSuite) SetupSuite() {
+	require.NotNil(ss.T(), ss.pool)
+	repo, err := repository.NewPostgresRepository(ss.pool)
+	require.NoError(ss.T(), err)
+	ss.repo = repo
+}
+
+func (ss *SyncNodesTestSuite) TearDownSuite() {
+	ss.pool.Close()
+}
+
+func (ss *SyncNodesTestSuite) TestSyncNodes() {
+
+	ctx := context.Background()
+	nowNano := time.Now().UnixNano()
+
+	tests := []struct {
+		name          string
+		stateNodes    []*dao.NodesDAOInfo
+		existingNodes []*model.Node
+		expectedNodes []*model.Node
+		wantErr       bool
+	}{
+		{
+			name: "Sync nodes with no existing nodes",
+			stateNodes: []*dao.NodesDAOInfo{
+				{
+					PartitionName: "default",
+					Nodes: []*dao.NodeDAOInfo{
+						{ID: "1", NodeID: "node-1", Partition: "default", HostName: "host-1"},
```

Comment:
Should the `Partition`s here be `PartionID` instead?

Reply:
Updated after merge